framework/tst/dslabs/framework/testing/runner/SimulatedNetwork.java

```diff
@@ -0,0 +1,202 @@
+package dslabs.framework.testing.runner;
+
+import dslabs.framework.Address;
+import dslabs.framework.testing.Event;
+import dslabs.framework.testing.MessageEnvelope;
+import dslabs.framework.testing.TimerEnvelope;
+import dslabs.framework.testing.runner.Network.Inbox;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.PriorityQueue;
+import java.util.Random;
+import javax.annotation.Nonnull;
+import lombok.extern.java.Log;
+
+/**
+ * A deterministic simulated network.
+ */
+// the inheritance is for easily reusing `RunState`. consider extract common
+// behavior of `Network` and `SimulatedNetwork` into interface(s)
```

Comment:
Yes, absolutely there should be an interface. It doesn't look like you're reusing any of the `Network.java` logic.